src/com/aaronzalewski/EmailCrawler.java

```diff
@@ -0,0 +1,37 @@
+package com.aaronzalewski;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Set;
+
+public class EmailCrawler {
+
+    private static Set<String> emailAddresses;
+
+    public static void main(String[] args) {
+        emailAddresses = new HashSet<>();
+
+        breadthFirstSearchCrawl("http://" + args[0]);
+
+        emailAddresses.forEach(System.out::println);
```

Comment:
although it's functionally cleaner, one problem here is that there's no output along the way.